### PR TITLE
cilium/cmd: Remove deprecated policy_trace command

### PR DESCRIPTION
**Pull request by sayboras**
Signed-off-by: Tam Mach <tam.mach@cilium.io>

### CODE REVIEW

File: cilium/cmd/policy_trace.go [hunk 1]
```diff
@@ -1,271 +0,0 @@
...
-						Labels: v,
-					},
-					To: &models.TraceTo{
-						Labels: w,
-						Dports: dPorts,
-					},
-					Verbose: verbose,
-				}
-
-				params := NewGetPolicyResolveParams().WithTraceSelector(&search).WithTimeout(api.ClientTimeout)
```
Comment (gandro) [inline cilium/cmd/policy_trace.go:132]: If we remove the command, we can also remove the `WithTraceSelector` infrastructure here. Should we do it in this PR? Or do we have a tracking issue for that follow-up?

Comment (sayboras) [reply cilium/cmd/policy_trace.go:132]: yup, let me clean it up in this PR as well :+1:

Comment (ldelossa) [reply cilium/cmd/policy_trace.go:132]: Wanna put into draft for now @sayboras ?

**Review by gandro — APPROVED** [review]
Looks good. I believe there there is potentially even more code to be removed, i.e. some of the code here: https://github.com/cilium/cilium/blob/5feb34e83a73aead156373fd812fbc6afb42d575/pkg/policy/policy.go#L17-L28


But it might be easier to remove that in a separate PR

**Review by ldelossa — APPROVED** [review]
LGTM

**Comment by sayboras** [comment]
/test-1.16-4.19

**Comment by sayboras** [comment]
test-runtime hits https://github.com/cilium/cilium/pull/23297

**Review by aditighag — APPROVED** [review]

**Comment by sayboras** [comment]
/test-1.16-4.19

**Comment by sayboras** [comment]
test-1.16-4.19 hits https://github.com/cilium/cilium/issues/23845

**Comment by sayboras** [comment]
test-1.16-4.19 hits https://github.com/cilium/cilium/issues/23845 :( 

https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-kernel-4.19/56/testReport/junit/Suite-k8s-1/16/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/

```
DNS entry is not ready after timeout
Expected
    <*errors.errorString | 0xc002282480>: {
        s: "DNS 'app1-service.default.svc.cluster.local' is not ready after timeout: 7m0s timeout expired",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-K8s-1.16-kernel-4.19/src/github.com/cilium/cilium/test/k8s/updates.go:374
```

**Comment by sayboras** [comment]
Required code owner reviews are in, CI are all passed. Marking this ready to merge.

**Pull request merged**